### PR TITLE
br: Fail backup if any snapshots failed

### LINKED ISSUE #53037: Snapshot backup stuck when EBS snapshot failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Create a `VolumeBackup` (either adhoc or via schedule)
2. If underlying EBS snapshot encounters issue, then data plane backup job for that snapshot will be stuck with log

```
2024-04-29, 23:06:09.295		
I0430 06:06:09.295599       9 backup.go:302] [2024/04/30 06:06:09.295 +00:00] [INFO] [ebs.go:241] ["check pending snapshots"] [count=3]
...
```

### 2. What did you expect to see? (Required)

backup job, `Backup`, and `VolumeBackup` should fail. We don't expect these failures to resolve (in fact there is separate status for recoverable errors)

### 3. What did you see instead (Required)

backup job runs indefinitely until manual cleanup

### 4. What is your TiDB version? (Required)

v6.5.4

### PR DESCRIPTION
**Pull request by michaelmdeng**
### What problem does this PR solve?

During EBS snapshot backup, if any snapshots fail, the backup will be stuck indefinitely checking the snapshots that failed. Since we only run a single backup at a time, this also blocks subsequent backups.

Issue Number: close #53037 

Problem Summary: Backup stuck when EBS snapshot failed

### What changed and how does it work?

Update snapshot backup process to fail if it detects snapshots in failed state. We observe rare snapshot failures due to underlying EBS hardware issues.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/aws/ebs.go [hunk 1]
```diff
@@ -251,6 +251,9 @@ func (e *EC2Session) WaitSnapshotsCreated(snapIDMap map[string]string, progress
 			if *s.State == ec2.SnapshotStateCompleted {
 				log.Info("snapshot completed", zap.String("id", *s.SnapshotId))
 				totalVolumeSize += *s.VolumeSize
+			} else if *s.State == ec2.SnapshotStateError {
+				log.Error("snapshot failed", zap.String("id", *s.SnapshotId), zap.String("error", (*s.StateMessage)))
+				return 0, errors.Errorf("snapshot %s failed", *s.SnapshotId)
```
Comment (nkg-) [inline br/pkg/aws/ebs.go:256]: Looks great Backup metadata is still generated, and contains the snapshot ids. So cleanup should delete the snapshots. https://github.com/pingcap/tidb/blob/ede35854d5b4ef165d90ac927b2ec89f7509b226/br/pkg/task/backup_ebs.go#L228-L255

Comment (nkg-) [reply br/pkg/aws/ebs.go:256]: I wonder though, how much time does it take for AWS to mark the snapshots as failed.

**Review by YuJuncen — APPROVED** [review]

**Review by BornChanger — APPROVED** [review]

**Comment by nkg-** [comment]
@BornChanger : Is the test failure relevant.

**Comment by michaelmdeng** [comment]
> @BornChanger : Is the test failure relevant.

Can I get a retest? updated to fix lint errors

**Comment by nkg-** [comment]
Please cherrypick to release-6.5 and release-7.5

**Pull request merged**